### PR TITLE
[collectd 6] Load plugin: Migrate to collectd 6 data structures.

### PR DESCRIPTION
**Pull request by octo**
ChangeLog: Load plugin: Migrated to collectd 6 data structures.

### CODE REVIEW

File: src/load.c [hunk 1]
```diff
@@ -83,30 +84,29 @@ static void load_submit(gauge_t snum, gauge_t mnum, gauge_t lnum) {
...
 
-  vl.values = values;
-  vl.values_len = STATIC_ARRAY_SIZE(values);
-
-  sstrncpy(vl.plugin, "load", sizeof(vl.plugin));
-  sstrncpy(vl.type, "load", sizeof(vl.type));
-
   if (cores > 0) {
-    sstrncpy(vl.type_instance, "relative", sizeof(vl.type_instance));
+    fam.name = "system.load.scaled";
```
Comment (eero-t) [inline src/load.c:95]: IMHO reporting differently named metric should be a user decision, not one done by plugin itself.

**Review by eero-t — COMMENTED** [review]
Changes look fine, but I have different opinion on metric naming decision.

File: src/load.c (hunk 1 above)
Comment (octo) [reply src/load.c:95]: This is a bit hidden by the diff, but it *is* a user decision: line 81 evaluates `report_relative_load` and only sets `cores` if it is true. `report_relative_load` represents a configuration option.

Comment (eero-t) [reply src/load.c:95]: Ah, that's why it was doing `cores > 0` which I was wondering a bit about... What machine has zero CPUs? :-)

**Review by eero-t — APPROVED** [review]

**Pull request merged**